hardware_in_the_loop/software/modules/ecus/ecu.py

```diff
@@ -0,0 +1,34 @@
+from modules.other.iocontroller import IOController
+
+class ECU:
+    def __init__(self, io: IOController, states: dict, simulated: bool=False):
+        self.io = io
+        self.states = states  # This may be parsed from a DBC
+        self.simulated = simulated
+
+    def set_state(self, state: str, value):
+        """Set the state of a certain value of a simulated board
+
+        For example, because the Dashboard emits the Ready to Drive signal,
+        you could use `set_state("RTD", True)`
+
+        Args:
+            state (str): The state you want to set on the simulated ECU
+
+            value (multiple types): True or False for digital values, floats for 
+            analog values, ints for CAN values. This is how the iocontroller knows
+            which type of signal we want to set!
+
+        Returns:
+            None
+
+        Note: For digital values, True is "high" or "1", False is "low" or "0".
+        For analog values, always pass in the requested voltage; the conversion
+        happens downstream. For CAN values, pass the value of the byte you are
+        setting (0-255). TODO handle signals that take more than one byte, like
```

Comment:
Looking at this TODO, some background: `set_state` will call the IOController's `update` function, which sends a hardware signal. If we are updating a 2-byte signal, it is possible we send a new CAN message after only one of the bytes has been changed (which, for example, could mean we are requesting torque that we don't want, which could be bad).

One way to resolve this might be to include a `no_update` and a `force_update` option, so the IO controller only sends a signal to hardware once 2 or more bytes have been updated. Not super automatic, but not a bad option to deal with the few edge cases.